libraries/vendors/compile-altera.sh

```diff
@@ -513,7 +513,63 @@ if [ $STOPCOMPILING -eq 0 ] && [ "$COMPILE_ARRIA" == "TRUE" ] && [ $SKIP_LARGE_F
 
 	GHDLCompilePackages
 fi
+# compile Cyclone library
+if [ $STOPCOMPILING -eq 0 ] && [ "$COMPILE_CYCLONE" == "TRUE" ]; then
+	Library="cyclone"
+	Files=(
+		cyclone_atoms.vhd
+		cyclone_components.vhd
+	)
 
+	# append absolute source path
+	SourceFiles=()
+	for File in ${Files[@]}; do
+		#Don't put nonexisting files.
+		if [ -f "$SourceDirectory/$File" ]; then
```

Comment:
Don't duplicate code like `$SourceDirectory/$File`. Create a new variable and use it in the existence check and the item append operation.